damageext/damageext.c

```diff
@@ -679,13 +680,13 @@ PanoramiXDamageCreate(ClientPtr client, xDamageCreateReq *stuff)
 static int
 PanoramiXDamageDelete(void *res, XID id)
 {
-    int i;
     PanoramiXDamageRes *damage = res;
 
-    FOR_NSCREENS_BACKWARD(i) {
-        if (damage->damage[i]) {
-            DamageDestroy(damage->damage[i]);
-            damage->damage[i] = NULL;
+    unsigned int walkScreenIdx;
+    FOR_NSCREENS_BACKWARD(walkScreenIdx) {
```

Comment:
Will this ever stop?
I'm not a fan of using an unsigned int for iterators. This can easily lead to a problem like [this](https://stackoverflow.com/questions/9044059/unsigned-integers-in-c-for-loops) and such an infinite loop is quite easy to overlook.

Reply:
you're right. it cannot be unsigned yet, as long as we're using the old macro here. the new replacement (still in some WIP queues) aren't counting below zero anymore (using a 2nd index variable that's subtracted by 1 instead). Overlooked this when factoring the name change into preparational queues.

background: the new macros will look a bit like lambda expression, which automatically declares the walk variable on its own (locally scoped), also declares a pointer to the actual screen, and they will always have the same name. Those PRs here are just doing the name change separately, so the diffs getting smaller.

Reply:
@metux thanks for the context, it makes more sense now :)